src/modules/fancyzones/editor/FancyZonesEditor/Models/Settings.cs

```diff
@@ -418,24 +421,58 @@ private void ParseCommandLineArgs()
             WorkArea = SystemParameters.WorkArea;
 
             string[] args = Environment.GetCommandLineArgs();
-            if (args.Length == 6)
+
+            if (args.Length == 2)
+            {
+                if (args[1].Equals("Debug"))
+                {
+                    const string debugDeviceInfoJson =
+                        @"{
+                        ""device-id"": ""BOE06C6#4&a213ba1&0&UID265988_1920_1080_{03891D5D-CC32-4FD9-9B94-846FE43F4690}"",
+                        ""active-zoneset"": {
+                            ""uuid"": ""{C82EBE61-9234-49BA-8392-09963C2C4CED}"",
+                            ""type"": ""columns""
+                        },
+                        ""editor-show-spacing"": true,
+                        ""editor-spacing"": 16,
+                        ""editor-zone-count"": 4
+                    }";
+
+                    ParseDeviceInfoData(debugDeviceInfoJson);
+                }
+                else
+                {
+                    string title = "FancyZones Editor Error";
+                    MessageBox.Show("FancyZones Editor arguments are invalid.", title);
+                    ((App)Application.Current).Shutdown();
+                }
+            }
+            else if (args.Length == 3)
             {
                 var parsedLocation = args[(int)CmdArgs.X_Y_Width_Height].Split('_');
+                if (parsedLocation.Length != 4)
+                {
+                    string title = "FancyZones Editor Error";
+                    MessageBox.Show("FancyZones Editor arguments are invalid.", title);
+                    ((App)Application.Current).Shutdown();
+                }
+
                 var x = int.Parse(parsedLocation[0]);

```

Comment:
why are these hardcoded index values?

Reply:
This is how we pack and pass the workarea size to Editor (x_y_width_height) from the engine. I can introduce named indexes for readability purposes..

I also tried to eliminate this argument as well, but for some reason calling `GetMonitorInfo` from C# (by using DllImport) returns workarea size with no scaling included, and calling it from C++ it does include scaling - which is what we need.

Reply:
I think You need to set DPI awareness for process, before calling functions that might be impacted with it, which is definitelly `GetMonitorInfo`. Check https://docs.microsoft.com/en-us/windows/win32/api/shellscalingapi/nf-shellscalingapi-setprocessdpiawareness

Reply:
Could be. I'll try it. Thanks!